src/console_reporter.cc

```diff
@@ -44,22 +47,39 @@ bool ConsoleReporter::ReportContext(const Context& context) {
       color_output_ = false;
   }
 #endif
-  std::string str = FormatString("%-*s %13s %13s %10s\n",
-                             static_cast<int>(name_field_width_), "Benchmark",
-                             "Time", "CPU", "Iterations");
-  GetOutputStream() << str << std::string(str.length() - 1, '-') << "\n";
 
   return true;
 }
 
+void ConsoleReporter::PrintHeader(const Run& run) {
+  std::string str = FormatString("%-*s %13s %13s %10s",
+                                 static_cast<int>(name_field_width_),
+                                 "Benchmark", "Time", "CPU", "Iterations");
+  if(!run.counters.empty()) {
+    str += " UserCounters...";
```

Comment:
@EricWF @dominichamon note the use of the ellipsis, as an indication that the output below is not vertically aligned. Example results:

```
-----------------------------------------------------------------------------------------------
Benchmark                                         Time           CPU Iterations UserCounters...
-----------------------------------------------------------------------------------------------
BM_UserCounter/threads:8                       2266 ns      10283 ns      68080 Bar=16 Bat=40 Baz=24 Foo=8
BM_UserCounter/threads:1                       9791 ns       9785 ns      71503 Bar=2 Bat=5 Baz=3 Foo=1024m
BM_UserCounter/threads:2                       4915 ns       9829 ns      71076 Bar=4 Bat=10 Baz=6 Foo=2
BM_UserCounter/threads:4                       2603 ns      10283 ns      68256 Bar=8 Bat=20 Baz=12 Foo=4
BM_UserCounter/threads:8                       2152 ns      10263 ns      68440 Bar=16 Bat=40 Baz=24 Foo=8
BM_UserCounter/threads:16                      1484 ns      10283 ns      68224 Bar=32 Bat=80 Baz=48 Foo=16
BM_UserCounter/threads:32                       906 ns      10285 ns      68096 Bar=64 Bat=160 Baz=96 Foo=32
BM_UserCounter/threads:4                       2605 ns      10279 ns      68140 Bar=8 Bat=20 Baz=12 Foo=4
```